controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -145,19 +145,21 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		return ctrl.Result{Requeue: true}, nil
+		return ctrl.Result{}, err
 	}
 
 	// Fetch the Cluster.
 	cluster, err := util.GetOwnerCluster(ctx, r.Client, kcp.ObjectMeta)
 	if err != nil {
-		log.Error(err, "Failed to retrieve owner Cluster from the API Server")
-		return ctrl.Result{}, err
+		if !apierrors.IsNotFound(err) {
```

Comment:
I'm not sure I agree with this change.
The fact that the we get apierrors.IsNotFound(err) when reading the cluster is an actual issue to be investigated, so we should return error.

Instead when we get cluster == nil but no error, it means that the owner ref to the cluster is not yet set, and it is ok to return becaue another reconcile will be triggered as soon as the owner ref is set

Reply:
Okay. I agree with your opinion.
I apply the commit https://github.com/kubernetes-sigs/cluster-api/pull/10907/commits/11bb972a4e9d17c1c424af6e23d874c4d47614eb.